test/call.c

```diff
@@ -3402,4 +3402,130 @@ int test_call_sni(void)
 
 	return err;
 }
+
+
+int test_call_cert_select(void)
+{
+	int err = 0;
+	struct fixture fix, *f = &fix;
+	char auri_tls[256], buri_tls[256];
+	const char *dp = test_datapath();
+	char s[256];
+
+	/* Set valid global certificate. */
+	re_snprintf(conf_config()->sip.cert, sizeof(conf_config()->sip.cert),
+		    "%s/sni/server-interm.pem", dp);
+	conf_config()->sip.verify_server = false;
+	conf_config()->sip.verify_client = true;
+
+	TEST_ERR(err);
+
+	fixture_init(f);
+
+	mem_deref(f->a.ua);
+	mem_deref(f->b.ua);
+
+	f->behaviour = BEHAVIOUR_ANSWER;
+
+	re_snprintf(s, sizeof(s), "A <sip:a@127.0.0.1;transport=tls>"
+		    ";regint=0;cert=%s/sni/client-interm.pem", dp);
+	err = ua_alloc(&f->a.ua, s);
+	TEST_ERR(err);
+
+	re_snprintf(s, sizeof(s), "B <sip:b@127.0.0.1;transport=tls>"
+		    ";regint=0;cert=%s/sni/other-cert.pem", dp);
+	err = ua_alloc(&f->b.ua, s);
+	TEST_ERR(err);
+
+	re_snprintf(auri_tls, sizeof(auri_tls), "sip:a@127.0.0.1:%u",
+		    sa_port(&f->laddr_tls));
+	re_snprintf(buri_tls, sizeof(buri_tls), "sip:b@127.0.0.1:%u",
+		    sa_port(&f->laddr_tls));
+
+	/* 1st test. No CA set. Call from A to B. TLS handshake must fail. */
+	f->b.n_closed = 1;
+
+	err = ua_connect(f->a.ua, 0, NULL, buri_tls, VIDMODE_OFF);
+	TEST_ERR(err);
+
+	err = re_main_timeout(5000);
+	TEST_ERR(err);
+	TEST_ERR(fix.err);
+
+	ASSERT_EQ(0, fix.a.n_incoming);
+	ASSERT_EQ(0, fix.a.n_established);
+	ASSERT_EQ(1, fix.a.n_closed);
+	ASSERT_EQ(0, fix.a.close_scode);
+
+	ASSERT_EQ(0, fix.b.n_incoming);
+	ASSERT_EQ(0, fix.b.n_established);
+	ASSERT_EQ(1, fix.b.n_closed);
+	ASSERT_EQ(0, fix.a.close_scode);
```

Comment:
Is it correct to use `fix.a.close_code` here ? should it be `b` ?

Reply:
Thank you! Fixed.